From e292ec0120d27e87de9aed864e4ca425c6da6194 Mon Sep 17 00:00:00 2001
From: Emergi Idan <idan.emergi@intel.com>
Date: Wed, 22 Jul 2020 00:16:52 +0300
Subject: [PATCH] PONRTSYS-7979:
 ppa-sessions-are-not-getting-timeout-after-crawler-is-disabled-in-pp

Export API to get number of opened sessions
---
 drivers/net/ppv4/pp-api/pp-session-mgr/pp_session_mgr.c | 13 +++++++++++++
 include/linux/pp_api.h                                  |  7 +++++++
 2 files changed, 20 insertions(+)

diff --git a/drivers/net/ppv4/pp-api/pp-session-mgr/pp_session_mgr.c b/drivers/net/ppv4/pp-api/pp-session-mgr/pp_session_mgr.c
index 736f17c4388a..c1ced4438572 100644
--- a/drivers/net/ppv4/pp-api/pp-session-mgr/pp_session_mgr.c
+++ b/drivers/net/ppv4/pp-api/pp-session-mgr/pp_session_mgr.c
@@ -1448,6 +1448,19 @@ s32 pp_max_sessions_get(u32 *max_sessions)
 }
 EXPORT_SYMBOL(pp_max_sessions_get);
 
+s32 pp_open_sessions_get(u32 *open_sessions)
+{
+	if (unlikely(ptr_is_null(open_sessions)))
+		return -EINVAL;
+
+	if (unlikely(!pp_is_ready()))
+		return -EPERM;
+
+	*open_sessions = atomic_read(&db->stats.sess_open);
+	return 0;
+}
+EXPORT_SYMBOL(pp_open_sessions_get);
+
 s32 pp_session_stats_get(u32 id, struct pp_stats *stats)
 {
 	struct pp_dsi dsi;
diff --git a/include/linux/pp_api.h b/include/linux/pp_api.h
index 3eeb6eed9868..6412dbdf540b 100755
--- a/include/linux/pp_api.h
+++ b/include/linux/pp_api.h
@@ -1034,6 +1034,13 @@ s32 pp_hostif_del(struct pp_hostif_cfg *hif);
  */
 s32 pp_max_sessions_get(u32 *max_sessions);
 
+/**
+ * @brief get the pp number of open sessions
+ * @param open_sessions number of open sessions
+ * @return s32 return 0 for success
+ */
+s32 pp_open_sessions_get(u32 *open_sessions);
+
 /**
  * @brief get session statistics
  * @note Cannot be used from irq context
